### ISSUE TITLE
Regression in @babel/plugin-transform-named-capturing-groups-regex with unicode regexps

### ISSUE THREAD

**Issue by Jessidhia** [babel/babel#10601]
## Bug Report

**Current Behavior**
Likely a regression from https://github.com/babel/babel/pull/10430, but the contents of `u` regexps are getting transformed to being non-`u` without updating a regexp's flags. The problem becomes a more serious bug when the `s` flag is also present, as the transformed `.` no longer matches newlines.

**Input Code**
- [REPL link](https://babeljs.io/repl#?babili=false&code_lz=MYewdgzgLgBATgUxgXhgegDwAoD8GoCGA5gHwA6AJgJQDUJAdAFR5lpkDW-xJJaEArgChQkEABsE9MSCJZE9KAmhYA5BgAMJAB46yYAJ6GMaTSqpV0aGBAAWIfmIowARkihx-CIA&presets=&prettier=true&version=7.6.4&externalPlugins=%40babel%2Fplugin-transform-named-capturing-groups-regex%407.6.3)

```js
const re = /<(?<tag>\d)+>.*?<\/\k<tag>>/su
console.log(re.test('<0>xxx\nyyy</0>')) // should be true
```

- [Link using an older version of the plugin, without the regression](https://babeljs.io/repl#?babili=false&code_lz=MYewdgzgLgBATgUxgXhgegDwAoD8GoCGA5gHwA6AJgJQDUJAdAFR5lpkDW-xJJaEArgChQkEABsE9MSCJZE9KAmhYA5BgAMJAB46yYAJ6GMaTSqpV0aGBAAWIfmIowARkihx-CIA&presets=&prettier=true&version=7.6.4&externalPlugins=%40babel%2Fplugin-transform-named-capturing-groups-regex%407.4.5)

**Expected behavior/code**
A clear and concise description of what you expected to happen (or code).

Output with `@7.4.5` (ignoring helpers, just the regexp):

```js
/<(\d)+>.*?<\/\1>/su
```

Output with `@7.6.3`:

```js
/<([0-9])+>(?:[\0-\t\x0B\f\x0E-\u2027\u202A-\uD7FF\uE000-\uFFFF]|[\uD800-\uDBFF][\uDC00-\uDFFF]|[\uD800-\uDBFF](?![\uDC00-\uDFFF])|(?:[^\uD800-\uDBFF]|^)[\uDC00-\uDFFF])*?<\/\1>/su
```

Note how the regexp, which didn't even have to be transformed _from_ being a unicode regexp (and didn't remove the `u` flag after doing it), specifically skips over both `\x0A` and `\x0D`, the `\n` and `\r` characters, which are supposed to be matched because the regexp has the `s` flag.

**Babel Configuration (.babelrc, package.json, cli command)**

```js
{
  "plugins": ["plugin-transform-named-capturing-groups-regex"]
}
```

**Environment**
- Babel version(s): 7.6.4
- Node/npm version: v12.10.0 / yarn 1.19.1
- OS: macOS 10.15
- Monorepo: yes
- How you are using Babel: `babel-jest`

**Possible Solution**
Need some way to tell `regexpu` to not do the unicode downleveling; or if unicode transformation is "inevitable" the `dotAll` flag needs to be respected.

**Comment by nicolo-ribaudo** [comment]
Need to check if it's fixed by https://github.com/babel/babel/pull/10447

**Comment by JLHwung** [comment]
Will be fixed in #10447, see https://github.com/babel/babel/pull/10447#issuecomment-545966579 for current approach.

> Need some way to tell regexpu to not do the unicode downleveling

True, I will propose a `useDotAllFlag` to regexpu-core.
